### ISSUE TITLE
Every fitted model should report a standard fit-stats summary by default (like OLS reports R²)

### ISSUE THREAD

**Issue by nealcaren** [nealcaren/topica#806]
## Motivation

When you fit an OLS regression, you get a standard block back without asking: N, R², adjusted R², F, log-likelihood, AIC/BIC. You do not go hunting for each number. A fitted topic model in topica has no equivalent. The numbers exist, but they are scattered across per-model attributes, and which ones exist depends on the model:

```python
m = topica.LDA(20).fit(c)
m.log_likelihood   # LDA: yes
m.perplexity       # LDA: yes
m.converged        # everywhere
m.fit_history      # everywhere

topica.NMF(20).fit(c).log_likelihood   # AttributeError — NMF has no likelihood
topica.CTM(20).fit(c).perplexity       # AttributeError
```

So a user cannot write one line that works across the roster to see "how did this fit go, and how good is it." They have to know each model's surface.

## Proposal

Give every fitted model a `.summary()` (and a `__repr__`/`_repr_html_` that shows a compact version), returning one standard block, with `N/A` where a stat does not apply to that family rather than an `AttributeError`. A first cut of the standard set:

- **Shape**: K (num_topics), D (num docs), V (vocab size)
- **Fit**: converged (bool), iterations run, wall time if recorded
- **Objective**: log-likelihood / ELBO / reconstruction error (whichever the family optimizes), and perplexity where defined
- **Quality** (intrinsic, corpus-free): topic diversity, mean coherence (UMass fallback when no reference corpus is passed), mean exclusivity
- **Provenance**: model name, seed

`summary(texts=None)` so coherence can use a real reference corpus when the user has one, and fall back to the model's own UMass otherwise (same contract as `evaluate.diagnostics`).

## Relationship to what exists

`topica.evaluate.diagnostics(model)` already gives a per-**topic** table (coherence, exclusivity, FREX, size, ...). This issue is the per-**model** scalar summary that sits above it: the one-glance header, not the topic breakdown. `diagnostics` stays the drill-down.

## Design questions

- Return type: a small dataclass with `.to_markdown()`/`_repr_html_` (matches the `render()` house style used by manifest/compare), or a plain dict? Dataclass is friendlier for the repr.
- Should `.summary()` compute coherence/diversity eagerly (a cost on every repr) or lazily/cached? Repr should be cheap, so the auto-repr shows only stored stats (converged, iters, ll/elbo, K/D/V) and `summary(texts=)` adds the quality metrics on request.
- Cross-family objective naming: one `objective` field labeled by family (`log_likelihood` / `elbo` / `reconstruction_error`), or separate columns with N/A? Labeled single field reads cleaner.
- Standardize the underlying attributes too (e.g. a common `n_iter`/`num_iterations`, `num_docs`) so the summary has a uniform surface to read, per the naming-conventions contract.

Filed from the #786 recon follow-up work.

**Comment by nealcaren** [comment]
## Design decided (2026-08-20)

**Two tiers.** The OLS-`R²` analogy only holds for the model-agnostic quality metrics, because the per-family objective is not comparable across families (Gibbs log-likelihood vs variational ELBO vs NMF reconstruction error vs embedding models with no likelihood). So the summary separates:

- **Fit diagnostics (family-internal, comparable only within a family):** `converged`, `iterations`, the objective, `perplexity` where defined.
- **Quality metrics (model-agnostic, the real cross-family "R²"):** coherence, diversity, exclusivity — computable for every model from topic-word (± a reference corpus).

**Default surface: rich repr, cheap stats only.** A fitted model's `__repr__`/`_repr_html_` auto-shows the *stored* scalars (converged, iters, objective, K/D/V) so `print(m)` / a bare notebook eval gives the block "by default," like OLS. The quality tier (coherence needs a reference corpus and is not free) is computed only on an explicit `m.summary(texts=...)`, matching the `evaluate.diagnostics` contract. Repr never pays a coherence cost.

**Objective presentation: named columns + n/a.** Separate `log_likelihood` / `elbo` / `reconstruction_error` fields, each `n/a` where inapplicable, rather than one relabeled "objective" field. More explicit; a reader sees exactly which objective a family optimizes and that the others do not apply. `perplexity` likewise `n/a` outside the Gibbs/likelihood families.

**Relationship to existing surface.** `evaluate.diagnostics(model)` stays the per-**topic** table (coherence, exclusivity, FREX, size, ...). `.summary()` is the per-**model** header above it. Return a small dataclass with `_repr_html_`/`to_markdown()` in the same house style as `compare`/manifest `render()`.

**Scope note.** 55 models. The summary *reads* whatever fit surface a model exposes and marks `n/a` gracefully; it does not force every family to invent a likelihood. Standardizing the underlying attribute names (a common `iterations`, `num_docs`, etc.) is folded in under the existing naming-conventions contract (`test_naming_conventions.py`).

Next: turn this into an implementation plan (dataclass shape + the per-family objective/perplexity resolver + the repr wiring + a conformance test that every model produces a valid summary).

**Comment by nealcaren** [comment]
## Open items carried over from #809 (sample-user audit of the fit-summary)

#809 is closed in favor of this issue; its unchecked items are the remaining to-do list here. The LDA spike with the Tier-1 held-out guard and Tier-4 labels is preserved at tag `parked/fit-summary-lda-spike` (aca85db).

- [ ] **`make_heldout()` split scored by a model NOT fit on `ho.documents` is accepted silently — contaminated `heldout_loglik`.** ✓verified (`-17.66`, no guard). `eval_heldout` requires the model be fit on `heldout.documents`; nothing checks it, so a model fit on the full corpus (which saw the held-out tokens) still reports a log-likelihood. A correct guard needs the model to retain a training fingerprint — a **Rust-side change**, deferred. Until then, `make_heldout`/`eval_heldout` docstrings should state the ordering (split *before* fit) even more loudly.
- [ ] **Which held-out row you get depends silently on the *type* passed, and it's backwards from trust.** The *legal* `make_heldout` path populates `heldout_loglik` and leaves `perplexity (held-out) = n/a`; the *illegal* raw-training-docs path populated `perplexity`. So the safe path shows no perplexity and the unsafe path did. Consider computing both for a `Heldout`, or documenting in-line why one is shown.
- [ ] **Rows carry no units/estimator.** `perplexity (held-out)` (≈227, lower better, document-completion) vs `heldout_loglik` (≈ −27, higher better, stm-style mean per-doc word-holdout) — the summary never says which estimator or unit. For a paper the unit is mandatory. Suggest `perplexity (held-out, document-completion)` and `heldout_loglik (mean per-doc)`.
- [ ] **Two LDA perplexities exist; the summary uses the weaker one.** `topica.perplexity`'s own docstring notes LDA also has the Wallach et al. left-to-right estimator (`LDA.perplexity`/`LDA.evaluate`). A user reporting "perplexity" won't know which one the header is.
- [ ] **Raw strings to `heldout=` raise a misleading error:** `"none of the held-out tokens were in the model vocabulary"`. Real cause: the strings weren't tokenized/pruned to the model vocab. Detect `list[str]` and tokenize, or say "held-out looks un-tokenized." (Now partly masked by the Tier-1 guard, which requires `assume_unseen=True` first.)
- [ ] **No document-split helper for the perplexity path.** `make_heldout` produces the *word*-holdout / `heldout_loglik` path; getting the honest document-completion perplexity meant splitting the DataFrame, refitting, and hand-tokenizing the test set by hand. Consider a `train_test_split(corpus)` or a `mode="documents"` on `make_heldout`.
- [ ] **The five topic-health fields are undocumented.** ✓verified: `help()` on `diversity`, `topic_redundancy`, `topic_significance`, `weak_topics`, `effective_topics` returns `"The type of the None singleton"` (they're `float | None` dataclass fields). No range, direction, or threshold — Claude would omit all five from a paper. Add a one-line docstring per field (range, direction, source).
- [ ] **`converged = no` is effectively always "no" for Gibbs LDA and alarms first-timers.** ✓verified: collapsed-Gibbs LDA defaults `convergence_tol=0.0` (never early-stops), so the repr shows `no` by design. Suggest `n/a (fixed N sweeps)` for Gibbs models, or show the last delta. (Arguably a topica-wide repr issue, not fit-summary's alone.)
- [ ] **Two things named `summary`.** `topica.summary(model)` = top words; `model.summary()` = fit-stats. Mild trap; the only reason Claude found the latter was the `(call .summary(...))` hint in the repr.
